feat: use sha256 to check file integrity

luojiyin1987 opened the pull request.

luojiyin1987 committed:
feat: use sha256 to check file integrity

luojiyin1987 committed:
feat: check window zip

luojiyin1987 committed:
feat: parse_checksum_file

luojiyin1987 committed:
feat: more error info && get user tips

luojiyin1987 committed:
warn info friendly

luojiyin1987 committed:
code clear

luojiyin1987 committed:
fix: sha256 error

luojiyin1987 commented:
@lispking   Please check the review

stefan-mysten commented:
@luojiyin1987 thanks for this PR. I will have a look asap.

jnaulty reviewed (changes requested):
Great feature addition to the `suiup` `install.sh` script.
I have some requested changes  (primarily merging `download_checksum` + `download_file`)

luojiyin1987 committed:
fix: code clear

luojiyin1987 committed:
refactor: download_checksum

luojiyin1987 committed:
refactor: verify_download_integrity

stefan-mysten commented on install.sh:
```diff
@@ -15,6 +15,18 @@ NC='\033[0m' # No Color
+# in trusted environments or when you can manually verify the downloaded binary.
+
+# Global flag to track if integrity verification was skipped
+INTEGRITY_CHECK_SKIPPED=false
```
nit: I wonder if it makes sense to rename this
```suggestion
SKIP_INTEGRITY_CHECK=false
```

stefan-mysten commented on install.sh:
```diff
@@ -230,14 +525,51 @@ install_suiup() {
     
+    # Special handling for Windows x86_64 (not currently available in releases)
+    if [ "$os" = "windows" ] && [ "$arch" = "x86_64" ]; then
+        printf '%bWarning: Windows x86_64 is not currently available. Only ARM64 is supported.%b\n' "${YELLOW}" "${NC}"
```
This needs to be fixed, this x86_64 is supported.

luojiyin1987 committed:
fix: windows version info note

luojiyin1987 committed:
fix: INTEGRITY_CHECK_SKIPPED to SKIP_INTEGRITY_CHECK

luojiyin1987 committed:
fix: no support windows x86_64

stefan-mysten commented:
> I'm sorry , did not click `show all` <img alt="image" width="966" height="144" src="https://private-user-images.githubusercontent.com/6524977/474104779-0c195363-3ed3-44de-8b2f-d84027e31308.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMjE5MTMsIm5iZiI6MTc1NDMyMTYxMywicGF0aCI6Ii82NTI0OTc3LzQ3NDEwNDc3OS0wYzE5NTM2My0zZWQzLTQ0ZGUtOGIyZi1kODQwMjdlMzEzMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDRUMTUzMzMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjU5YjY1OTBjOWU2MmMxZjQ3YjdlOWFhNjY3ZjA1MzllMDI3YzdjM2ZhY2VkYjg1NjQ0NmY4N2UyYjhjNDE0NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.XGTSyCPd7bdn97ZnLsXQez7jl0q5zrp8eJ5oJFdaoVM">

No worries at all, nothing to be sorry about!

luojiyin1987 committed:
fix: sh compatibility

luojiyin1987 committed:
doc: update readme

luojiyin1987 committed:
feat: improve checksum verification error handling for MD5

- Make MD5 verification consistent with SHA256 behavior
- Treat malformed checksum files as skipped verification instead of failure
- Add clear warning messages and skip reasons
- Improve user experience while maintaining security awareness

luojiyin1987 committed:
fix: check path in windows

luojiyin1987 committed:
doc: update readme

stefan-mysten commented on README.md:
````diff
@@ -32,11 +32,11 @@ With `suiup`, you can:
 ### From Script
 ```bash
-curl -sSfL https://raw.githubusercontent.com/Mystenlabs/suiup/main/install.sh | sh
+curl -sSfL https://raw.githubusercontent.com/Mystenlabs/suiup/main/install.sh | bash
````
Why to change to bash?

luojiyin1987 replied:
local is a keyword specific to bash, used to declare local variables. There is no local keyword in standard sh.
local  ist nicht Teil des POSIX-Shell-Standards.

 On Linux distributions (such as CentOS, Fedora, RHEL, openSUSE, macOS), /bin/sh is indeed a symbolic link (symlink) pointing to /bin/bash. So in most cases, there is no problem. 

In Debian and its derivatives (such as Ubuntu), /bin/sh does not point to bash by default, but rather to a lighter-weight shell that more strictly adheres to POSIX standards—dash. This differs from bash support.

In order to detect problems earlier, it is recommended to use bash.

luojiyin1987 replied:
You can use `shellcheck` to check compatibility between different shells  @stefan-mysten

iM4M-nugrh reviewed (commented):
send me teacher for app github please
